photon-core/src/main/java/org/photonvision/common/configuration/SqlMigrations.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) Photon Vision.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.photonvision.common.configuration;
+
+/**
+ * Add migrations by adding the SQL commands for each migration sequentially to this array. DO NOT
+ * edit or delete existing SQL commands. That will lead to producing an icompatible database.
+ *
+ * <p>You can use multiple SQL statements in one migration step as long as you separate them with a
+ * semicolon (;).
+ */
+public final class SqlMigrations {
+    public static final String[] SQL = {
```

Comment:
Clever. Would an enum make it any clearer rather than inducing into an array? Or a map of version numbers to sql queries? Just a nit, since the version->index is now like a bit implicit.

Reply:
It might. I originally structured it as a switch statement, which was very explicit, but that felt clunky, so I moved to the array version. I thought about an Enum or mapping, but it seemed like that was going back in the direction of the switch statement. It would be more explicit, but it may also suggest a degree of freedom that we don't really have with the migrations. The version numbers aren't really something the programmer controls. You just have to assign it the next value when you add a new migration. That's why I sort-of added numbers with the comments. Someone reading the code can see the intent, but it isn't part of the actual data structure.

A mapping would be more useful if we added a table for storing the schema version. Then, we could specify semantic versioning for the migrations. The downside would be more complexity for relatively little gain.  

That being said, I'm game to try a different approach if you think that it would make it easier for future contributors to understand the code and do the right thing.

Reply:
As long as you think what we do now is future proof for the kinds of migrations we will potentially need to do to add new features (thinking things like saving ML models as binary blobs and adding new pipeline types and such) then I'm game to do this

Reply:
"Future proof" is a pretty tall ask, especially since this is the first time I've written any kind of database migration code. I do think that it's better than what we have now, but it may have failure modes that aren't obvious to me. As I see it, this approach has the following benefits and risks:

Benefits:
- It addresses an existing failure mode with database changes and provides a framework to handle future changes to the database.
- The code is pretty simple and self-contained. I think it will be easy for other contributors to understand and use correctly.
- It doesn't rely on external libraries or migration scripts. One less dependency that could lead to problems in the future.
- If used properly, it should make it easy to maintain the databases and will provide clear logged messages to help track down the error if there is a problem.

Risks:
- I'm very inexperienced at database maintenance and this approach may be naive and not work the way we need it to in the future.
- It may not be able to handle edge cases the way that more developed migration frameworks do.
- It only supports SQL-based changes to the database schema. If we ever need to make changes to the database that require supporting Java code, we'll have to incorporate that somehow in the migration method. There's already one ugly example of this in there for handling the beta databases, and I hope that we don't have to do that again once we deploy this code.

There are also some loose ends:
- If migration fails to bring the database to the correct version, what do we do? Archive the database and create a new one? Shut down PV with an error message? Show the error on the UI?
- When it detects that the database is too new, it issues a warning, but tries to use it anyway. This will probably cause problems when data is read or written. Should we do more to handle this?
- Should we add a table that tracks the migration history, such as which version of PV updated the table? What about date/time of the migration. I'm not sure the date is set correctly on a coproc.
- What unit tests do we need to validate that it is working?
- With the full release, do we want to start from scratch with a new database, or do we want to try to support databases that were created by beta versions?

I think that the benefits of adding it outweigh the risks and that it should be able to handle the types of changes that I can anticipate making to the database. 

I'd absolutely like more feedback and testing of this approach. Do you think we'll get more feedback if we merge it as is while we continue to work on things like unit tests and clearing up the loose ends?